### ISSUE TITLE
Feature Request - Save group as template | Collapse group

### ISSUE THREAD

**Issue by CHollman82** [comfyanonymous/ComfyUI#1912]
I love the new feature to save node templates... it seems like it would be very easy and very useful to be able to right-click a group box and select "save as template" to create a template with that group and every node in it, including connections. I setup group boxes as if they were programming functions, using redirect nodes as input and output connections, and this would be extremely useful to me:

![image](https://github.com/comfyanonymous/ComfyUI/assets/3136566/f311836d-3656-4b96-af06-6ee20f2c9196)

Also, being able to collapse a group box like you can collapse a node box would be great as well.

**Comment by tonynoce** [comment]
Also it would be nice to add the functionality to select the group when doing a selection, since they get ignored.
But yeah the implementation of the templates with groups would be handy

**Comment by fogostudio** [comment]
bump for this, it would be really useful

**Comment by dinmakers** [comment]
Me too. Please.

**Comment by Jellybit** [comment]
Came here looking to see if this was ever addressed. Not yet I guess? Grouping is used in other functionality, like with rgthree's Fast Groups Muter, which is in my opinion, essential for multi-functional workflows. Groups are also so handy for visual organization, and selecting all associated nodes easily and without question as to whether you got them all or not. They most definitely should be part of templates, at least optionally.

**Comment by brbbbq** [comment]
I second this. The template function is handy, but being able to save them in groups would be a lot more convenient. The whole process of making groups, sizing them, titling them, and color coding them is pretty tedious, so any opportunity to avoid it seems really worthwhile.

**Comment by 123LiVo321** [comment]
Yeah, Save Group As Template is really missing. Am I overlooking something, or it wasn't adressed yet? It is possible to save nodes as a template and in each workflow used put them in their groups again and again.... but it would be nice to just load the template WITH a group, instead of each time re-do .... if it isn't adressed by comfy annonymous, did someone do it custom-node-ly...?

**Comment by DefinitlyEvil** [comment]
I second this! 
Also it'd be cooler if there's something like a library which I can save globally.

**Comment by DaveMayfield** [comment]
I absolutely agree. "Convert to Group Node" is a wonderful feature, but since we cannot re-use those groups in more than one workspace it isn't all that useful. However, the combination of "convert to group node" and the ability to save it as a template would be game changer.

**Comment by Patrick-81** [comment]
Wouldn't it be possible to save the group with its content and when importing it recover the group with its content ?
That might helps a lot to build "meta-nodes" with connectors pipe style.

**Comment by JorgeR81** [comment]
While we wait for this, there is workaround, with a node suit:

https://github.com/comfyanonymous/ComfyUI/discussions/2936#discussioncomment-8803615

**Comment by tarkansarim** [comment]
I was just wondering about it myself. +1 on this.

**Comment by BenDes21** [comment]
still not possible... ?

**Comment by jffaust** [comment]
A must-have!

**Comment by 123LiVo321** [comment]
Will be this thing (save a group) finally possible in the new UI version? please, PLEASE, PLEASEEEEEEEEEEE ... xD
